views/js/controller/Tools/assessmentActivity.js

```diff
@@ -196,7 +193,96 @@ define([
                         show: false
                     }
                 }
-            }).render();
+            })
+            .render();
+        } else {
+            activityGraph.refresh({
+                graphConfig: {
+                    data: {
+                        json: data
+                    },
+                    axis: {
+                        x: {
+                            tick: {
+                                format: $activityGraphInterval.val() === 'day' ? '%H:%M' : '%m-%d'
+                            },
+                            label: {
+                                text: $activityGraphInterval.val() === 'day' ? __('Hours') : __('Days')
+                            }
+                        }
+                    }
+                }
+            });
+        }
+    }
+
+    function updateDeliveryList(data) {
+        if (!$deliveryListDatatable) {
+            $deliveryListDatatable = $deliveryListContainer.datatable({
+                filter:                   false,
+                model:                    deliveryListModel,
+                paginationStrategyTop:    'none',
+                paginationStrategyBottom: 'none',
+                selectable:               true,
+                sortorder:                'asc',
+                sortby:                   'label'
+            }, data);
+        } else {
+            $deliveryListDatatable.datatable('refresh', data);
+        }
+    }
+
+
+    return {
+        start: function () {
+            var autoRefreshInterval;
+            var config = $('.activity-dashboard').data('config');
+            var poll;
+
+            poll = polling({
+                action: function () {
+                    request(url.route('assessmentActivityData', 'Tools', 'taoProctoring', { interval: $activityGraphInterval.val() }))
+                    .then(function (data) {
+                        updateUserActivity(data.assessment_activity);
```

Comment:
As commented in a previous review (sorry if it was not clear enough), each of the 4 "monitoring widgets" should be wrapped into their own components (extending the ui/component) and unit tested. 
Eventually, we should only have in the controller the initialization of the 4 components in the appropriate containers and their update after every polling interval.